RetailCoder.VBE/Refactorings/Rename/RenameEventHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Rubberduck.Common;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.UI;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Rubberduck.Refactorings.Rename
+{
+    public class RenameEventHandler : RenameHandlerBase
```

Comment:
This is why "Handler" is a bad name. Is this "Rename Event" handler, or "Rename Event Handler"?

It really feels like each implementation should actually be a specialized `IRefactoring` implementation:

````csharp
    public interface IRefactoring
    {
        void Refactor();
        void Refactor(QualifiedSelection target);
        void Refactor(Declaration target);
    }
````
 
The parameterless `void Rename()` would be the parameterless `Refactor()` overload - the `Refactor(QualifiedSelection)` and `Refactor(Declaration)` overloads can easily be implemented too, even though they'd be somewhat redundant; they'd have to `Debug.Assert` that the parameter points to a declaration of the correct type.

The original idea behind the 3 overloads of `IRefactoring` was that we had:

 - A way to refactor off a given arbitrary selection - e.g. *extract method* requires this.
 - A way to refactor a specific target `Declaration` - e.g. when a refactoring is called from a quick-fix and the target doesn't need to be acquired.
 - A "smart" overload that figures out what the target is.

That 3rd one is the questionable one: ideally we would make all refactoring-calling code figure out the target, and pass it to the refactoring: it shouldn't be the job of a refactoring class to work out its own target. In all cases, the refactoring needs to prompt for a new name and put the "model" together.

So if these "handlers" were to become `IRefactoring` implementations, their base class would be responsible for:

 - Implementing the `IRefactoring` interface
 - Prompting for the new name / creating the model

I think I'd make the parameterless overload `public`, the `QualifiedSelection` overload `virtual`, and the `Declaration` overload `abstract`. Then the `RenameModel` can be a `protected` get-only property, and there can be an `abstract` method that makes the error message customizeable per-implementation.